tools/verdict/com.ge.research.osate.verdict/src/com/ge/research/osate/verdict/aadl2vdm/Aadl2Vdm.java

```diff
@@ -480,10 +558,9 @@ else if(dataPort.isOut()) {
 						modeString = "out";
 					}
 
-
-			    	verdict.vdm.vdm_model.Port newPort = createVdmPort(portName, modeString, dataPort.getQualifiedName());
-
-			    	//Note: Not populating "type" for now
+					//fetching data type information
+					DataSubcomponentType dSubCompType = dataPort.getDataFeatureClassifier();
+			    	verdict.vdm.vdm_model.Port newPort = createVdmPort(portName, modeString, dataPort.getQualifiedName(), dSubCompType);
```

Comment:
These are not duplicate instances, the dataport is different in each scenario. Will change the function createVdmPort to take dataport as a parameter rather than passing multiple parameters.